packages/wrangler/src/d1/execute.tsx

```diff
@@ -240,10 +241,27 @@ async function executeLocally({
 
 	logger.log(`🌀 Loading DB at ${readableRelative(dbPath)}`);
 
-	const sqliteDb = await createSQLiteDB(dbPath);
-	const db = new D1Database(new D1DatabaseAPI(sqliteDb));
-	const stmts = queries.map((query) => db.prepare(query));
-	return (await db.batch(stmts)) as QueryResult[];
+	const db = new D1Gateway(new NoOpLog(), storage);
+	let results: D1SuccessResponse | D1SuccessResponse[];
+	try {
+		results = db.query(queries.map((query) => ({ sql: query })));
+	} catch (e: unknown) {
+		throw (e as { cause?: unknown })?.cause ?? e;
+	}
+	assert(Array.isArray(results));
+	return results.map<QueryResult>((result) => ({
+		results: (result.results ?? []).map((row) =>
+			Object.fromEntries(
+				Object.entries(row).map(([key, value]) => {
+					if (Array.isArray(value)) value = `[${value.join(", ")}]`;
+					if (value === null) value = "null";
+					return [key, value];
+				})
+			)
+		),
+		success: result.success,
+		meta: { duration: result.meta?.duration },
+	}));
```

Comment:
Is Miniflare3 the reason for the additional complexity or just less mocking due to the more realistic environment?